### DISCUSSION TITLE
How can I Isolate a number from a string?

### DISCUSSION THREAD

**Discussion by deathwishdave** [taocpp/PEGTL#225] [Q&A] [answered]
Please may I get some assistance, I am struggling to select a set of grammars to read the the following:

```
 ! The following Location and Design Day data are produced as possible from the indicated data source.
 ! Wind Speeds follow the indicated design conditions rather than traditional values (6.7 m/s heating, 3.35 m/s cooling)
 ! No special attempts at re-creating or determining missing data parts (e.g. Wind speed or direction)
 ! are done.  Therefore, you should look at the data and fill in any incorrect values as you desire.
  
 Site:Location,
  LONDON/GATWICK_GBR Design_Conditions,     !- Location Name
      51.15,     !- Latitude {N+ S-}
      -0.18,     !- Longitude {W- E+}
       0.00,     !- Time Zone Relative to GMT {GMT+/-}
      62.00;     !- Elevation {m}
 
```

and extract the strings and numbers, but ignore the comments (!)

I have the following so far:

```
	struct SiteLocation : pegtl::string<'S', 'i', 't', 'e', ':', 'L', 'o', 'c', 'a', 't', 'i', 'o', 'n'> {};
	struct Comma : pegtl::one< ',' > {};
	struct Location : pegtl::plus<pegtl::not_one<','>> {};
	struct Latitude : pegtl::plus< pegtl::digit > {};
	struct Longitude : pegtl::plus< pegtl::digit > {};
	struct SemiColon : pegtl::one< ';' > {};

	struct HeaderOne : pegtl::must<pegtl::until<SiteLocation>, Comma, Location, Comma, Latitude, Comma, Longitude, pegtl::until<SemiColon>> {};
```

but I am struggling to isolate the numeric values from the ascii comments.

**Comment by d-frey** [comment, marked as answer]
Your grammar is not skipping whitespace or comments, even without comments it will not work. Note that the PEGTL does not automatically skip whitespace (and newlines) for you, it has to be part of the grammar.

Check out some of our examples, maybe start with https://github.com/taocpp/PEGTL/blob/master/src/example/pegtl/parse_tree.cpp and see how we use `pad` and `space` to allow skipping whitespace. Then look for examples like `abnf2pegtl.cpp` where we also skip comments.

**Comment by deathwishdave** [comment]
Thanks very much.
